src/main/java/io/parsingdata/metal/data/selection/ByToken.java

```diff
@@ -37,4 +38,19 @@ public static ParseItem get(final ParseGraph graph, final Token definition) {
         return get(graph.tail, definition);
     }
 
+    public static ParseItemList getAll(final ParseGraph graph, final Token definition) {
+        if (definition == null) { throw new IllegalArgumentException("Argument definition may not be null."); }
+        return getAllRecursive(graph, definition);
+    }
+
+    private static ParseItemList getAllRecursive(final ParseGraph graph, final Token definition) {
```

Comment:
I didn't pass the empty ParseItemList all the way through, because it's only used as a base case return value and I find it slightly more clear as it is now.

I just noticed I should change the getAll call for the tailResults to getAllRecursive.